### ISSUE TITLE
WCT: Function to see competitions a user is registered for

### ISSUE THREAD

**Issue by LinusFresz** [thewca/worldcubeassociation.org#1723]
The WCT receives a lot of mails related to competitions, asking for event changes, adding guests to their registrations, cancelling the registration and much more. Unfortunately lots of these people do not tell us which competition they are referring to, nor tell us their name, WCA ID or anything else. So all we have is a mail address. To solve this issue, members of the WCT need to first respond asking for the competition the sender is asking about and then telling them whom to contact for further information or forward the mail to the organizers.
An easy way to prevent this would be some option for the WCT to see all registered competitions for a mail. This should be quite easy to implement as an individual user can already access the competitions he/she registered for via "My Competitions". I see different options how we could deal with this:

1. Add a list of all competitions a user registered for to their profile here: https://www.worldcubeassociation.org/users/7056/edit. This would allow the WCT to still use the "Manage users" tab to find users by name, ID or mail, but we could also see their upcoming competitions.
2. Add another list where we can see all registered competitions for a specific user/name/ID/mail

It would be great if this could be implemented and would help the WCT to save a lot of time answering mails!

**Comment by SAuroux** [comment]
Wow, you really spend so much time dealing with such people? Why not simply reply with a standard mail that tells them how to find the right addressee (<their competition> -> info -> contact)?

**Comment by LinusFresz** [comment]
Well, actually we do this! But you can’t imagine how many mails we receive from the contact form or forwarded from the board with the same questions over and over again (from the same people who do not use our advice but contact us again). This is also related to the issue that the contact link at the bottom is still linked to contact@ (and will remain how it is) instead to the organizers. See Jeremy’s arguments here: https://github.com/thewca/worldcubeassociation.org/issues/1531#issuecomment-294435006

**Comment by SAuroux** [comment]
Well, in the interest of keeping the WCT non-burned-out I would seriously advise to go a way that requires less effort per case:

- reask the WST to fix the contact links
- use standard mails (i.e. "here is the way to contact the right person" and/or "this is the information you need to give us")
- ignore people that take too much of your time by being too lazy/stupid despite several attempts to help them properly!

**Comment by LinusFresz** [comment]
I have to admit I do not really agree with your opinion. Personally I like the way how the contact link works. We take the risk to let the people contact us (or any other team) again if we do not give them (relatively) detailed information. So why not add this feature and make this easier for the WCT? It would not need much more time to look into the database and see which competitions a competitor will attend in the future, so we could reduce our mail contact to these people from two mails to only one and even give them detailed information on how they need to proceed! Also I do not agree to just ignore these people, this is not a nice way to deal with this and might cause even more frustration in the community or make them contact the board/another team again, which will also not solve the problem.

**Comment by SAuroux** [comment]
I don't really see how your reply corresponds to what I was saying. But it is perfectly fine if you do not agree with me. Let's just hope that you still feel the same way in 1-2 years. :)

I think that this feature should not be too hard to implement. However, in case this is not fully clear to you: it is only possible to find registrations of competitions using the WCA registration system.

**Comment by LinusFresz** [comment]
Yes, for now the implementation would be great! We can see how our/my opinion about this might change in the future and start this discussion again, if necessary.

> However, in case this is not fully clear to you: it is only possible to find registrations of competitions using the WCA registration system.

Thanks, I am aware of that. Due to the increasing amount of competitions using the WCA registration system, I am confident that the implementation will help us a lot. :-)

@thewca/software-team is this implementation which I proposed in my first comment something you could do for us?

**Comment by Laura-O** [comment]
Wouldn't this issue be (at least partly) solved by changing the contact form?

I'm not saying this shouldn't be implemented, just saying that it's probably more helpful to implement a change which will help people writing clear messages instead of implementing a change which will help you to understand unclear messages.

**Comment by LinusFresz** [comment]
I guess both things together would do the trick! I can definitely agree that a changed contact form will help, too. As you know we are already into discussions about this.

>Edit: so what you should do is agree on how the contact form should look like and ask the Software team for an implementation.

Will do soon! Unfortunately I have some other important stuff to do now, but I hope to get this done soon.

**Comment by rodrisan97** [comment]
I was thinking about the possibility of putting in the user's profile (public) a list of his upcoming competitions (which he is registered).
People that want to follow the competitions of a person could do it easily, now it is so tedious.

For example, I'd like to know what competitions Feliks will go, so I access to his profile and I can see it in a section like "Results" "Podiums" or "Map"

**Comment by jfly** [comment]
@rodrisan97 @UnsolvedCypher, let's try to keep this issue focused on actions to help the WCT. Please create a new issue to discuss publicly viewable registrations if you want to talk about what that would look like =)

I am down with the idea of building a page for the WCT to see competitions that a given user is registered for. The simpl@LinusFresz if you est possible implementation might be to extend the `/competitions/mine` page to take an optional user id, and maybe add a link to that from the user profile page.

I'll ping Lauren to see if this still feels like something the WCT would want.

**Comment by lamacl** [comment]
An update from the WCT on this: Even with modifications to the contact form (now you can choose which competition you have questions about and it will contact those organizers directly), we still get many emails through the other contact page options and through emails to notifications@, contact@, and through the board. I can see a feature of being able to see what competitions a person is currently registered for being useful in some situations, but in my experience, most of the questions that we get regarding competitions have to do with people asking questions before their registration has been approved (for example, how do I pay, why is my registration not approved, etc.) which I don't think this fix would be able to help.

In short, I think this feature would help us in some situations but it would not eliminate the need for us to send responses to people asking which competition they are referring to if they haven't fully registered for the competition they have questions about. If the WST has time to put towards this or it would be an easy feature, the WCT would be able to use it but it doesn't have to be placed as a priority at this point (in my opinion).

**Comment by dunkOnIT** [comment]
Having an internal-facing way to see competitions a user is signed up for would still be useful - I feel like I've seen this issue elsewhere though
